#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_ffmpeg.dpatch by Tobias Grimm <tg@e-tobi.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: FFMPEG related fixes

@DPATCH@
diff -urNad vdr-plugin-graphtft-0.3.2~rc2~/Makefile vdr-plugin-graphtft-0.3.2~rc2/Makefile
--- vdr-plugin-graphtft-0.3.2~rc2~/Makefile	2009-05-03 16:37:43.000000000 +0200
+++ vdr-plugin-graphtft-0.3.2~rc2/Makefile	2009-05-03 16:37:43.000000000 +0200
@@ -197,14 +197,6 @@
 ifdef HAVE_AVCODEC
   AVCODEC_INC = `pkg-config libavcodec --cflags`
 
-  ifeq (plain, $(shell pkg-config libavcodec --cflags | grep -q "ffmpeg" && echo plain))
-    DEFINES += -DPLAN_FFMPEG_INC
-  endif
-
-  ifeq (plain, $(shell pkg-config libavcodec --cflags | grep -q "libavutil" && echo plain))
-    DEFINES += -DPLAN_FFMPEG_INC
-  endif
-
   INCLUDES += $(AVCODEC_INC)
   LIBS += `pkg-config libavcodec --libs`
   DEFINES += -DHAVE_FFMPEG
diff -urNad vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/dvbrenderer/mpeg2encoder.c vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/dvbrenderer/mpeg2encoder.c
--- vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/dvbrenderer/mpeg2encoder.c	2009-05-03 16:37:36.000000000 +0200
+++ vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/dvbrenderer/mpeg2encoder.c	2009-05-03 16:37:57.000000000 +0200
@@ -26,7 +26,7 @@
 #ifdef PLAN_FFMPEG_INC
 #  include <avcodec.h>
 #else
-#  include <libavutil/avcodec.h>
+#  include <libavcodec/avcodec.h>
 #endif
 
 #ifdef HAVE_SWSCALE
@@ -134,11 +134,11 @@
    int i;
    AVPicture avpsrc;
 
-   avpicture_fill(&avpsrc, buf, PIX_FMT_RGBA32, width, height);
+   avpicture_fill(&avpsrc, buf, PIX_FMT_RGB32, width, height);
 
 #ifndef HAVE_SWSCALE
 
-   img_convert(&pic, PIX_FMT_YUV420P, &avpsrc, PIX_FMT_RGBA32, width, height);
+   img_convert(&pic, PIX_FMT_YUV420P, &avpsrc, PIX_FMT_RGB32, width, height);
 
 #else
 
diff -urNad vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/fbrenderer/fbrenderer.c vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/fbrenderer/fbrenderer.c
--- vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/fbrenderer/fbrenderer.c	2009-05-03 16:37:36.000000000 +0200
+++ vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/fbrenderer/fbrenderer.c	2009-05-03 16:38:01.000000000 +0200
@@ -35,7 +35,7 @@
 #ifdef PLAN_FFMPEG_INC
 #  include <avcodec.h>
 #else
-#  include <libavutil/avcodec.h>
+#  include <libavcodec/avcodec.h>
 #endif
 
 #include <common.h>
@@ -198,7 +198,7 @@
 
 	switch (fb_vinfo.bits_per_pixel) 
    {
-		case 32: tell(4, "fbdevout.c: using 32 bit depth"); fb_type = PIX_FMT_RGBA32; break;
+		case 32: tell(4, "fbdevout.c: using 32 bit depth"); fb_type = PIX_FMT_RGB32; break;
 		case 24: tell(4, "fbdevout.c: using 24 bit depth"); fb_type = PIX_FMT_RGB24;  break;
 		case 16: tell(4, "fbdevout.c: using 16 bit depth"); fb_type = PIX_FMT_RGB565; break;
 		default: tell(4, "fbdevout.c: color depth not supported -> %i bits per pixel",
diff -urNad vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/fbrenderer/mpeg2decoder.h vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/fbrenderer/mpeg2decoder.h
--- vdr-plugin-graphtft-0.3.2~rc2~/imlibrenderer/fbrenderer/mpeg2decoder.h	2009-05-03 16:37:36.000000000 +0200
+++ vdr-plugin-graphtft-0.3.2~rc2/imlibrenderer/fbrenderer/mpeg2decoder.h	2009-05-03 16:37:43.000000000 +0200
@@ -22,7 +22,7 @@
 #ifdef PLAN_FFMPEG_INC
 #  include <avcodec.h>
 #else
-#  include <libavutil/avcodec.h>
+#  include <libavcodec/avcodec.h>
 #endif
 }
 
